blackbox/handlers/storage/dropbox.py

```diff
@@ -0,0 +1,112 @@
+        self.client = DropboxClient(self.config.get("access_token"))
+
+    def sync(self, file_path: Path) -> None:
+        """Sync a file to Dropbox."""
+        # This is size what can be uploaded as one chunk.
+        # When file is bigger than that, this will be uploaded
+        # in multiple parts.
+        chunk_size = 4 * 1024 * 1024
+
+        upload_path = f"{self.upload_base}{file_path.name}"
+
+        try:
+            with file_path.open("rb") as f:
+                file_size = file_path.stat().st_size
+                if file_size <= chunk_size:
+                    self.client.files_upload(
+                        f.read(), upload_path, WriteMode.overwrite
+                    )
+                else:
+                    session_start = self.client.files_upload_session_start(
+                        f.read(chunk_size)
+                    )
+                    cursor = UploadSessionCursor(
+                        session_start.session_id,
+                        offset=f.tell()
+                    )
+                    # Commit contains path in Dropbox and write mode about file
+                    commit = CommitInfo(upload_path, WriteMode.overwrite)
+
+                    while f.tell() < file_size:
+                        if (file_size - f.tell()) <= chunk_size:
+                            self.client.files_upload_session_finish(
+                                f.read(chunk_size),
+                                cursor,
+                                commit
+                            )
+                        else:
+                            self.client.files_upload_session_append(
+                                f.read(chunk_size),
+                                cursor.session_id,
+                                cursor.offset
+                            )
+                            cursor.offset = f.tell()
+            self.success = True
+        except (ApiError, HttpError) as e:
+            log.error(e)
+            self.success = False
+            self.output = str(e)
+
+    def rotate(self) -> None:
+        """
+        Rotate the files in the Dropbox directory.
+
+        All files in base directory of backups will be deleted when they
+        are older than `retention_days`, and because of this,
+        it's better to have backups in isolated folder.
+        """
+        # Receive first batch of files.
+        files_result = self.client.files_list_folder(
+            self.upload_base if self.upload_base != "/" else ""
+        )
+        entries = [
+            e for e in files_result.entries if isinstance(e, FileMetadata)
+        ]
+
+        # If there is more files, receive all of them.
+        while files_result.has_more:
+            cursor = files_result.cursor
+            files_result = self.client.files_list_folder_continue(cursor)
+            entries += [
+                e for e in files_result.entries if isinstance(e, FileMetadata)
```

Comment:
```suggestion
                entry for entry in files_result.entries if isinstance(e, FileMetadata)
```
No single letter variables, please.